addons/hw_drivers/tools/helpers.py

```diff
@@ -283,10 +293,13 @@ def get_odoo_server_url():
         ap = subprocess.call(['systemctl', 'is-active', '--quiet', 'hostapd']) # if service is active return 0 else inactive
         if not ap:
             return False
-    return read_file_first_line('odoo-remote-server.conf')
+
+    return get_conf('remote_server')
```

Comment:
Error found:
Imagine an IoT Box in v16 migrating to v17

1) They connect the IoT Box to the new database. It's the code in v16 which will save the configuration of the associated database in `write_file('odoo-remote-server.conf', url)`
2) The IoT Box will then proceed to `git reset --hard` onto v17
3) When Odoo restarts, the check `server_url = helpers.get_odoo_server_url()` will return None
4) Since it will always return None we will never enter the check `if platform.system() == 'Linux' and server_url:`
5) So, the method `migrate_old_config_files_to_new_config_file()` will never be called